ansible/post-recheck-comment.yml

```diff
@@ -0,0 +1,68 @@
+---
+
+- hosts: localhost
```

Comment:
We have an idea to reduce complexity of this playbook. Idea requires a few steps:

1. Define gerrit server in inventory directly
    - we could create `inventory.production` and `inventory.development`
    - production variant would hold `review.opencontrail.org`
    - development variant would hold `review2.opencontrail.org`
2. We could extract keyfile creating into a separate script/playbook
    - we were thinking about emploing `yq` tool and use it like this

        ```bash
        ansible-vault view group_vars/gerrit/vault | yq -r ".gerrit_keys.${GERRIT_HOST}" > "${KEY_FILE}"
        ```

    - then we could run a playbook `post-recheck-comment.yml` (we're only not sure about `extra-args` part)

        ```
        ansible-playbook -i inventory.production \
            --private-key="${KEY_FILE}" \
            --ssh-extra-args="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" \
            post-recheck-comment.yml
        ```

    - this way `post-recheck-comment.yml` playbook would be responsible only for:
        - prompting for change id and patchset
        - running `post-recheck-comment` role

We checked `yq` tool availability and it kinda sucks because:

- it depends on `jq` tool, installable with apt
- `yq` can then be installed with `pip`

We have tested the keyfile extraction command from point 2. and it worked for following versions:

- `sudo apt install jq=1.5+dfsg-1`
- `pip install yq==2.6.0`

We can discuss IRL, but what do you think?

Reply:
Not sure about it. This indeed reduces scope of this playbook, but simultaneously adds some assumptions, moves some of the complexity to a separate script and makes it harder to run this playbook by hand.

> we could create `inventory.production` and `inventory.development`

This seems like a good idea. Is it possible to have the inventory sample, with group definitions, as we now have and create two files that "include" that inventory and just define gerrit servers?

> it depends on jq tool, installable with apt

Sidenote: Funny fact, I've used the `jq` for filtering jsons in the firsts iteration, but then rewritten using filters, because I didn't want to introduce a new dependency.